### ISSUE TITLE
function to output array of available ports

### ISSUE THREAD

**Issue by jordancluts** [JuliaIO/LibSerialPort.jl#39]
There currently isn't a way (that I can find) to obtain the list of ports. `list ports()` prints them to the REPL but doesn't return them in any way for a function to do further processing. Would it be possible to add a function or add this functionality to `list_ports`?

I'd be happy to help but I don't quite know how the internals with ptrs and refs works.

**Comment by andrewadare** [comment]
Out of curiosity, are you seeing this behavior on Windows? #22 is still unresolved.

**Comment by andrewadare** [comment]
Reading more carefully, I now see that this is different than #22.

**Comment by andrewadare** [comment]
Does this do what you want?
```Julia
using LibSerialPort


function get_port_list(;nports_guess::Integer=64)
    ports = sp_list_ports()
    port_list = String[]
    for port in unsafe_wrap(Array, ports, nports_guess, own=false)
        port == C_NULL && return port_list
        push!(port_list, sp_get_port_name(port))
    end
    sp_free_port_list(ports)
    return port_list
end

println(get_port_list())
```

**Comment by jordancluts** [comment]
Yes! that is precisely the functionality I was looking for. I can certainly use this function for now but this would be great to have added to the package by default.

**Comment by yakir12** [comment]
I too was looking for the same functionality. This seems like it should be included and exported. The main use-case is in a GUI where a user needs to choose a port from a list of available ones, or in reality, for the program to select the only viable one. 
Thanks!

**Comment by andrewadare** [comment]
Now that the PR is merged to master, can someone explain to me how to update the tag to 0.2.0? Sorry for my ignorance. I have not done this before.

**Comment by yakir12** [comment]
Me neither... I should learn though...

**Comment by yakir12** [comment]
OK, [apparently](https://julialang.github.io/Pkg.jl/v1/creating-packages/#Registering-packages-1), you should use [TagBot](https://github.com/apps/julia-tagbot).

**Comment by yakir12** [comment]
Yea, so I just did this, it's easy:
1. "install" Registrator and TagBot in this repo.
2. comment this:
```
 @JuliaRegistrator register
```
here

**Comment by yakir12** [comment]
^ yea, @andrewadare, you'll have to do it yourself.

**Comment by andrewadare** [comment]
@JuliaRegistrator register

**Comment by andrewadare** [comment]
@JuliaRegistrator register

**Comment by andrewadare** [comment]
As a first-timer, I found this release process confusing. 

LibSerialPort.jl is already registered, so yesterday I kept overlooking the "register" command in the Registrator docs, searching instead for some kind of "tag" or "release" command. 

I appreciate that granting write access to a bot to update Project.toml is not desirable, so it makes sense that I would need to modify it myself. But I didn't see why I need a web app or GitHub plugin to change a version number in a text file, so I was/am a little confused there.

Also, invoking `@JuliaRegistrator register` from an arbitrary issue comment was a bit unintuitive to me, but I certainly can't disagree with people who say it's easy to do.

**Comment by yakir12** [comment]
Yea, I agree. But it's pretty cool how automated and easy it is when everything just works...
